ec/src/hashing/curve_maps/swu.rs

```diff
@@ -24,38 +23,26 @@ pub trait SWUConfig: SWCurveConfig {
 /// Represents the SWU hash-to-curve map defined by `P`.
 pub struct SWUMap<P: SWUConfig>(PhantomData<fn() -> P>);
 
-/// Trait defining a parity method on the Field elements based on [\[1\]] Section 4.1
-///
-/// - [\[1\]] <https://datatracker.ietf.org/doc/draft-irtf-cfrg-hash-to-curve/>
-pub fn parity<F: Field>(element: &F) -> bool {
-    element
-        .to_base_prime_field_elements()
-        .find(|&x| !x.is_zero())
-        .map_or(false, |x| x.into_bigint().is_odd())
-}
-
 impl<P: SWUConfig> MapToCurve<Projective<P>> for SWUMap<P> {
     /// Checks if `P` represents a valid map.
     fn check_parameters() -> Result<(), HashToCurveError> {
         // Verifying that ZETA is a non-square
-        if P::ZETA.legendre().is_qr() {
-            return Err(HashToCurveError::MapToCurveError(
-                "ZETA should be a quadratic non-residue for the SWU map".to_string(),
-            ));
-        }
+        debug_assert!(
+            !P::ZETA.legendre().is_qr(),
+            "ZETA should be a quadratic non-residue for the SWU map"
+        );
 
         // Verifying the prerequisite for applicability  of SWU map
-        if P::COEFF_A.is_zero() || P::COEFF_B.is_zero() {
-            return Err(HashToCurveError::MapToCurveError("Simplified SWU requires a * b != 0 in the short Weierstrass form of y^2 = x^3 + a*x + b ".to_string()));
-        }
+        debug_assert!(!P::COEFF_A.is_zero() && !P::COEFF_B.is_zero(),
+		      "Simplified SWU requires a * b != 0 in the short Weierstrass form of y^2 = x^3 + a*x + b ");
```

Comment:
Should these checks be reverted to `if`s, or do we want to keep them as `debug_assert`s?

Reply:
This check only concerns the pre-computed parameters, there's nothing runtime dependant.